hazelcast/src/main/java/com/hazelcast/spi/tenantcontrol/DestroyEventContext.java

```diff
@@ -16,27 +16,21 @@
 
 package com.hazelcast.spi.tenantcontrol;
 
+import com.hazelcast.core.HazelcastInstance;
 import com.hazelcast.spi.annotation.Beta;
 
 /**
  * Hook to decouple Hazelcast object from the tenant
- * @param <T> context type
  */
 @Beta
-public interface DestroyEventContext<T> {
+public interface DestroyEventContext {
```

Comment:
even though this interface is marked `@Beta`, removing methods in a patch release is too much change. I would recommend for 3.12.9's PR, if absolutely necessary, then just add the `tenantUnavailable` method and keep existing ones as they are.

Reply:
The implementation of those methods is greatly simplified with these modifications. If we keep them as they are, they will be next to useless. I've gained some experience implementing this and it's much, much better and simpler this way.